vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -268,9 +269,16 @@ public void onGroupSubmitted(ExperimentalGroupSubmitEvent groupSubmitted) {
       AddExperimentalGroupResponse response = experimentInformationService.addExperimentalGroupToExperiment(
           experimentId,
           new ExperimentalGroupDTO(groupSubmitted.variableLevels(), groupSubmitted.sampleSize()));
-      if (response.responseCode() == ResponseCode.SUCCESS) {
-        loadExperimentalGroups();
-        groupSubmitted.source().close();
+      switch (response.responseCode()) {
+        case SUCCESS -> {
+          loadExperimentalGroups();
+          groupSubmitted.source().close();
+        }
+        case CONDITION_EXISTS -> {
+          InformationMessage infoMessage = new InformationMessage("A group with the same condition exists already.", "");
+          StyledNotification notification = new StyledNotification(infoMessage);
+          notification.open();
```

Comment:
To include more expressiveness: maybe extracting both derived actions in methods and with a semantically sound description what happens?

Also: what is `groupSubmitted.source().close()` doing?